stdlib/list.mli

```diff
@@ -141,6 +141,13 @@ val filter_map : ('a -> 'b option) -> 'a list -> 'b list
     @since 4.08.0
 *)
 
+val concat_map : ('a -> 'b list) -> 'a list -> 'b list
+(** [List.concat_map f l] gives the same result as
+    {!List.concat}[ (]{!List.map}[ f l)].
```

Comment:
Actually I spoke too soon FWIW I agree with @alainfrisch's comment [here](https://github.com/ocaml/ocaml/pull/8760#discussion_r306833362). Surely you took care not to say "is" but "gives the same result as" , but when I read this doc strings my gut feeling is that it will blow the stack on large inputs. I think that adding a "but is tail-recursive" at the end would do no harm.

Reply:
(I added an explicit mention of tail-recursiveness.)